### PR TITLE
Cesc/update recipe

### PR DESCRIPTION
**Pull request by FrancescElies**
bumps version, adds some fixes, a maintainer and a todo for the test section

### CODE REVIEW

**Comment by mrocklin** [comment]
Just curious, is Windows out of the question for blosc?

**Comment by jakirkham** [comment]
That's a good question for @msarahan .

**Comment by jakirkham** [comment]
The failure on Travis CI is due to the feedstock needing an update. Please [re-render](https://github.com/conda-forge/conda-smithy#re-rendering-an-existing-feedstock) with `conda-smithy` version `1.4.6` to address this. Can add `-c`/`--commit` to `conda smithy rerender` to commit the re-rendering.

**Comment by FrancescElies** [comment]
Thanks for the help, rerendering added missing appveyor file for windows

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,39 +1,44 @@
-{% set version = "1.3.2" %}
+{% set name = "blosc" %}
```
Comment (jakirkham) [inline recipe/meta.yaml:1]: Why the name change? Shouldn't this still be `python-blosc`? Note we do package [`blosc`](https://github.com/conda-forge/blosc-feedstock) also.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,39 +1,44 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
 
 requirements:
   build:
     - python
     - setuptools
-    - blosc
   run:
     - python
-    - blosc
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why are we dropping `blosc` as a dependency?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,39 +1,44 @@
...
     - numpy
 
 test:
-  imports:
+  requires:
     - blosc
```
Comment (jakirkham) [inline recipe/meta.yaml]: 😕

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,39 +1,44 @@
...
   commands:
-    - python -c "import blosc; blosc.test()"
+    - mkdir empty
+    - cd empty
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why is this needed?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,39 +1,44 @@
...
+    - python -c 'import blosc; blosc.print_versions()'
+    - python -c 'import blosc; blosc.test()'
 
 about:
   home: https://github.com/Blosc/python-blosc
   license: Apache 2.0
+  license_file: LICENSES/PYTHON-BLOSC.txt
```
Comment (jakirkham) [inline recipe/meta.yaml]: 👍

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,39 +1,44 @@
...
   commands:
-    - python -c "import blosc; blosc.test()"
+    - mkdir empty
+    - cd empty
+    - python -c 'import blosc; blosc.print_versions()'
+    - python -c 'import blosc; blosc.test()'
 
 about:
   home: https://github.com/Blosc/python-blosc
   license: Apache 2.0
```
Comment (jakirkham) [inline recipe/meta.yaml]: We could also optionally add `license_family: Apache`.

**Comment by jakirkham** [comment]
Ah, yeah, if you change `skip`, then re-rendering is needed to update the CI files included.

File: recipe/meta.yaml (hunk 1 above)
Comment (FrancescElies) [reply recipe/meta.yaml:1]: When you install one wants to run it you run, `import blosc` I thought this would be less confusing, but the repo is called `python-blosc`. Changing the name back to python-blosc
e0f075475ca69043d9b417db241020246d3dfd3e

File: recipe/meta.yaml (hunk 2 above)
Comment (FrancescElies) [reply recipe/meta.yaml]: I was doubting on this, I could not understand why do we need the package itself as a dependency?
1725c1a43a22c8569524876716c120849330a6fe

File: recipe/meta.yaml (hunk 3 above)
Comment (FrancescElies) [reply recipe/meta.yaml]: 69da7496e66c402768f80c561fe68195e28825f4

File: recipe/meta.yaml (hunk 4 above)
Comment (FrancescElies) [reply recipe/meta.yaml]: In my opinion we don't need it, I was just following the travis.ci file from the python-blosc project. so removing it
390e29d9238df4d5ed4ce5b3d22491b8ae147455

File: recipe/meta.yaml (hunk 6 above)
Comment (FrancescElies) [reply recipe/meta.yaml]: I believe I still need some input for the license, apparently there are some plans to change it to bsd https://github.com/Blosc/python-blosc/issues/121#issuecomment-258086960

Adding it acbca9268641fba1a514633dd8b928d903fbede4

File: recipe/meta.yaml (hunk 2 above)
Comment (danielfrg) [reply recipe/meta.yaml]: `blosc` is refering to the `c-blosc` lib, the recipe for that one is here: https://github.com/conda-forge/blosc-feedstock. This one `python-blosc` is just the python wrapper. We decided on those names to have the same ones as the `defaults` channel, here is the full discussion: https://github.com/conda-forge/staged-recipes/pull/918

Comment (danielfrg) [reply recipe/meta.yaml]: So `python-blosc` will build `c-blosc` if its not found (in the setup.py) but if we add it as a dependency it will just link it to the `c-blosc` that we build.

File: README.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
-About python-blosc
```
Comment (danielfrg) [inline README.md]: I guess that is is because of the rerendering of the feedstock? Maybe you need to rerender again.

**Comment by danielfrg** [comment]
+1 Great to see a maintainer come in for this one :D

File: README.md (hunk 7 above)
Comment (jakirkham) [reply README.md]: Correct. After fixing the name we will need another re-render. Note can do `conda smithy rerender --commit` to commit the changes. Also can tack on `auto` to commit the re-rendering automatically without reviewing the changes.

File: recipe/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipe/meta.yaml]: I guess the question will arise how we build `c-blosc`/`blosc` on Windows.

@msarahan can you please advise us on this? 😄

**Comment by jakirkham** [comment]
Could you please disable CircleCI on your fork @FrancescElies ? It seems to be building this PR on your fork instead of on this feedstock.

**Comment by jakirkham** [comment]
Also please close and reopen this PR after you have done that. This will trigger CircleCI to correctly build on this PR.

**Comment by FrancescElies** [comment]
I deactivated Circle Ci, Travis and Appveyor from my profile

**Comment by FrancescElies** [comment]
By the way, thank you both for your help!

**Comment by FrancescElies** [comment]
Reopened to trigger CI services

File: recipe/meta.yaml [hunk 8]
```diff
@@ -1,39 +1,44 @@
...
   number: 0
-  script: python setup.py install --blosc=$PREFIX --single-version-externally-managed --record record.txt
-  skip: True  # [win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
 
 requirements:
   build:
     - python
     - setuptools
-    - blosc
```
Comment (FrancescElies) [inline recipe/meta.yaml]: Shall I add this build requirements again?
The git history is a bit messy, once this PR is fine I could clean the history if you like.

**Comment by msarahan** [comment]
We have a c-blosc recipe internally, but I don't see packages, and I don't understand why.  Internally, I think we decided to change the compiled library to c-blosc, because that matches the project name.

File: recipe/meta.yaml (hunk 8 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Yes, let's go ahead and do that.

Sure if you want to clean it up at the end that sounds like a good idea.

**Comment by FrancescElies** [comment]
Now I just remembered why I changed the name of the package, from `python-blosc` to `blosc`
If I try to download `https://pypi.io/packages/source/p/python-blosc/python-blosc-1.4.4.tar.gz` it cannot be found
But if I try to this one`https://pypi.io/packages/source/b/blosc/blosc-1.4.4.tar.gz` seems to be fine.

File: recipe/meta.yaml [hunk 9]
```diff
@@ -1,39 +1,44 @@
-{% set version = "1.3.2" %}
+{% set name = "python-blosc" %}
+{% set version = "1.4.4" %}
+{% set sha256 = "76ae6e1cecddb320b5567ba78ee9a2ee31895ecba838fdf05fdb9131939a4705" %}
 
 package:
-  name: python-blosc
+  name: {{ name|lower }}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Maybe we should make this `python-{{ name|lower }}`.

File: recipe/meta.yaml [hunk 10]
```diff
@@ -1,39 +1,44 @@
-{% set version = "1.3.2" %}
+{% set name = "python-blosc" %}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Maybe we should change this to `blosc`.

File: recipe/meta.yaml [hunk 11]
```diff
@@ -1,39 +1,44 @@
...
 package:
-  name: python-blosc
+  name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://github.com/Blosc/python-blosc/archive/v{{ version }}.tar.gz
-  sha256: 63d1cd1da14087fa69bffc513b25d59deb24f45c65e4406acafaaa6ec9bd7cf2
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml:11]: This then should work.

**Review by jakirkham — COMMENTED** [review]
Naming is causing us some issues with the download. As it is called `blosc` on PyPI, but we call it `python-blosc` here, we need to make a few corrections.

**Comment by jakirkham** [comment]
Seems our comments passed each other. My comments should address the download issue.

File: recipe/meta.yaml (hunk 9 above)
Comment (FrancescElies) [reply recipe/meta.yaml]: sounds good to me

**Comment by jakirkham** [comment]
So I think this is mostly good now. The one issue is that we still don't have a `blosc` package for Windows. We have a few options.
1. Try to build a `blosc` package on Windows and come back to this.
2. Go back to skipping Windows.
3. Supply `blosc` on Unix and let Windows build `blosc` with the Python package.

Not sure which will be best.

**Comment by FrancescElies** [comment]
I'm not sure, to me the cleanest sounds (1.), but maybe for the moment we could skip windows. I'm easy

**Comment by FrancescElies** [comment]
I actually have some questions
Before we were using `python setup.py install --blosc=$PREFIX --single-version-externally-managed --record record.txt` to install `python-blosc` but I could not see in the project where the `PREFIX` for `blosc` is being set.
The current status of the recipe does not use `PREFIX` therefore when we install `python-blosc` it will use the `blosc` sources that are being shipped with `python-blosc`
Then, do we sill need blosc as dependency?
In `python-blosc` the recommended way is compiling with an installed Blosc library setting `BLOSC_DIR` env var, what shall we do?

**Comment by danielfrg** [comment]
`PREFIX` is set by conda-build. If the recommended way is to install blosc, link and then use `BLOSC_DIR` instead of `--blosc=...` we should change it to use that. What should the value of `BLOSC_DIR` the path to the compiled binaries?

**Comment by FrancescElies** [comment]
`--blosc=$PREFIX` has the same effect as setting `BLOSC_DIR`. 
It should be set to the path where blosc is being installed, from what I can see https://github.com/conda-forge/blosc-feedstock/blob/master/recipe/build.sh#L6 is being set to `PREFIX`, it seems like it should work.
I'll give it a try

**Comment by jakirkham** [comment]
> I'm not sure, to me the cleanest sounds (1.), but maybe for the moment we could skip windows. I'm easy

Me too. 😄 Would love to get it to work on Windows at some point though. 😉

**Comment by FrancescElies** [comment]
It looks like something wen't wrong now, 
`Error: HTTPError: 503 Server Error: Service Unavailable: Back-end server is at capacity for url: https://conda.anaconda.org/conda-forge/linux-64/openblas-0.2.18-6.tar.bz2: https://conda.anaconda.org/conda-forge/linux-64/openblas-0.2.18-6.tar.bz2`
I'll give a try to blosc on windows next week

**Comment by FrancescElies** [comment]
regtriggering CI services

**Comment by mrocklin** [comment]
Recommend retriggering CI services again

**Comment by FrancescElies** [comment]
Retriggered (close-open PR). I see appveyor and travis running again but not circle ci

**Comment by FrancescElies** [comment]
an empty commit retriggered circle ci (I'll remove it afterwards)

**Comment by FrancescElies** [comment]
removed empty commit,  + rerendering, fingers crossed

**Comment by FrancescElies** [comment]
tests passed now
